src/components/CardBlog/CardBlog.jsx

```diff
@@ -0,0 +1,92 @@
+import { Pagination } from '@mui/material'
+import { useEffect, useRef, useState } from 'react'
+import * as styleMui from './CardBlog.styled'
+import CardBlogList from './CardBlogList/CardBlogList'
+
+function CardBlog({ data, valueSearch, positions }) {
+    const [dataBlog, setDataBlog] = useState(data)
+    const [currentPage, setCurrentPage] = useState(1)
+    let itemsPerPage = 6
+
+    const getPosition = useRef()
+
+    // hàm nhận giá trị thừ tags để render ra bài viết
+    const searchSting = () => {
+        if (valueSearch !== '' || valueSearch.length > 0) {
+            const filteredData = data.filter((item) => {
+                return item.tags.some((tag) =>
+                    tag.tagName.includes(valueSearch)
+                )
+            })
+            setCurrentPage(1)
+            setDataBlog(filteredData)
+        } else {
+            setDataBlog(data)
+        }
+    }
+
+    // kích hoạt việc click vào các tags
+    useEffect(() => {
+        let searching = true
+        if (searching) {
+            searchSting()
+        }
+        // clean up function
+        return () => {
+            searching = false
+        }
+    }, [valueSearch])
+
+    // xử lý các giá trị scroll
+    const handleScroll = () => {
+        positions.scroll = window.pageYOffset
+        positions.sectionHeight = getPosition.current.clientHeight
+        positions.topContent = getPosition.current.offsetTop + 272
```

Comment:
What is the number `272`? You should create a variables for this number easy-to-read.